backend/src/main/java/com/google/blackswan/mock/SimpleAnomalyGenerator.java

```diff
@@ -0,0 +1,142 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.blackswan.mock;
+
+import static java.util.stream.Collectors.toMap;
+
+import com.google.models.*;
+import java.util.Scanner;
+import java.util.*;
+import java.text.ParseException;
+import java.util.logging.Logger;
+import java.io.IOException;
+import java.time.format.DateTimeParseException;
+import java.lang.Math;
+import java.lang.Object;
+
+/** Generate list of anomalies based on data in the csv file using average and threshold. */
+public class SimpleAnomalyGenerator implements AnomalyGenerator {
+  private static final String DATA_FILE_LOCATION = "/sample-ramen-data.csv";
+  private static final String EXCEPTION_MESSAGE = "Invalid row format.";
+  private static final Logger log = Logger.getLogger(SimpleAnomalyGenerator.class.getName());
+  private static final int THRESHOLD = 13;
+  private static final int NUM_POINTS = 5;
+
+  private List<Anomaly> anomalies;
+  private Map<Timestamp, Integer> data;
+
+  public SimpleAnomalyGenerator() {
```

Comment:
Also consider static factory methods instead of constructors for your implementations.

See effective java item 1 for more reading.

Reply:
Yep, I read the chapter about `static` factory methods and one of the benefit I think that could be applicable to `SimpleAnomalyGenerator` is _unlike constructors, they are not required to create new object each time they're invoked_.

Since current `SimpleAnomalyGenerator` generates the same list of anomalies each time, there is no need to `parseCSV()` and `generateAnomalies()` for every instantiation of `SimpleAnomalyGenerator`.  Do you think it is suitable to use static factory methods instead of constructor and return the same `SimpleAnomalyGenerator` each time? (Not sure but I think this would be kind of like a `singleton` class?)

My thoughts: I guess the drawback of this would be that the `SimpleAnomalyGenerator` would only be able to read from one csv file and always return the anomalies from that one csv file and this would also make testing more difficult as we cannot pass in a shorted version of csv file in form of a string to see if `SimpleAnomalyGenerator` generates the correct anomalies...

Reply:
Now, that i've refactored a lot of code, I don't think it is good for `SimpleAnomalyGenerator` to always return the same instance, since sometimes I want `SimpleAnomalyGenerator` to use a different threshold or to look at different data.

So feel free to ignore what I typed above :)